test/benchmarks/netperf/nptest.go

```diff
@@ -0,0 +1,660 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ nptest.go
+
+ Dual-mode program - runs as both the orchestrator and as the worker nodes depending on command line flags
+ The RPC API is contained wholly within this file.
+*/
+
+package main
+
+// Imports only base Golang packages
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"net/rpc"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type Point struct {
+	mss       int
+	bandwidth string
+	index     int
+}
+
+var mode string
+var port string
+var host string
+var worker string
+var kubenode string
+var podname string
+
+var workerStateMap map[string]*workerState
+
+var iperf_tcp_output_regexp *regexp.Regexp
+var iperf_udp_output_regexp *regexp.Regexp
+var netperf_output_regexp *regexp.Regexp
+
+var dataPoints map[string][]Point
+var dataPointKeys []string
+var datapointsFlushed bool
+
+var global_lock sync.Mutex
+
+const (
+	WORKER_MODE         = "worker"
```

Comment:
WorkerMode or workerMode - depending on the desired visibility. The same for other constants.